src/formats/xbm.zig

```diff
@@ -0,0 +1,201 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const ImageUnmanaged = @import("../ImageUnmanaged.zig");
+const FormatInterface = @import("../FormatInterface.zig");
+const color = @import("../color.zig");
+const utils = @import("../utils.zig");
+const buffered_stream_source = @import("../buffered_stream_source.zig");
+
+/// XBM is a monochrome bitmap format in which data is stored as a C language data array.
+/// Primarily used for the storage of cursor and icon bitmaps for use in the X graphical user interface.
+// In place of the usual image file format header, XBM files have two or four #define statements. The first two #defines specify the height and width of the bitmap in pixels. The second two specify the position of the hotspot within the bitmap, and are not present if no hotspot is defined in the image.
+// The labels of each #define contain the name of the image. Consider an image that is 8x8 pixels in size, named FOO, with a hotspot at pixel 0,7. This image contains the following #define statements:
+// #define FOO_width 8
+// #define FOO_height 8
+// #define FOO_x_hot 0
+// #define FOO_y_hot 7
+// The image data itself is a single line of pixel values stored in a static array. Data representing our FOO image appears as follows:
+//
+// Static unsigned char FOO_bits[] = {
+// 0x3E, 0x80, 0x00, 0x7C, 0x00, 0x82, 0x41, 0x00};
+//
+// Because each pixel is only one bit in size, each byte in the array contains the information for eight pixels, with the first pixel in the bitmap (at position 0,0) represented by the high bit of the first byte in the array. If an image width is not a multiple of eight, the extra bits in the last byte of each row are not used and are ignored.
+// XBM files are found in two variations: the older XI 0 format and the newer (as of 1986) XI 1 format. The only difference between these formats is how the pixel data is packed. The XI 1 flavor stores pixel data as 8-bit BYTEs. The older XI 0 flavor stores pixel data as 16-bit WORDs. There are no markers separating the rows of image data in either of these formats, and the size of an XBM array is limited only by the compiler and machine using the bitmap.
+// The XI 0 XBM is considered obsolete. Make sure that any X software you write is able to read both the XBM XIO and XI 1 formats, but when you write data, use only the XI 1 XBM format.
+pub const XBM = struct {
+    width: u32,
+    height: u32,
+    hotspot_x: u32,
+    hotspot_y: u32,
+    pixels: []const u8,
```

Comment:
The pixels is never used and should not be there, we never store pixels data in format implementation